### DISCUSSION TITLE
Has HTMLLoader made hierarchical display adjustments?

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#3031] [Q&A]
I remember in earlier versions, the HTMLLoader object should be able to control the display hierarchy, allowing Sprite to be displayed on top of it.
But now I can't seem to change it by setting the addChild method, as HTMLLoader always comes before any displayed objects.

```
import flash.html.HTMLLoader;
import flash.display.Sprite;

var html: HTMLLoader = new HTMLLoader();
var urlReq: URLRequest = new URLRequest("http://www.baidu.com/");
html.width = stage.stageWidth;
html.height = stage.stageHeight;
html.load(urlReq);
addChild(html);


var spr:Sprite = new Sprite();
addChild(spr);
spr.graphics.beginFill(0x0990099);
spr.graphics.drawCircle(0,0,100);
spr.graphics.endFill();
```
Or is there any way to make webView appear at the bottom level.
Of course, it is feasible to create a new form, but there will be an efficiency problem. When I set the form transparent, it will become too laggy. If transparency is not set, the interaction will become very hard because there is no way to make an image or animation from 0 to 100 transparency

**Comment by itlancer** [comment]
`HTMLLoader` (using legacy WebKit) deprecated with new AIR versions. Under the hood it uses `StageWebView`. So you cannot overlay it by anything in simple way.
You can try to create `StageWebView` in new `NativeWindow` and display this window above. Or rasterize `StageWebView` content to `BitmapData` and display it in `DisplayList` (but you lose interactivitiy of web page).
Discussed here:
https://github.com/airsdk/Adobe-Runtime-Support/issues/1428
https://github.com/airsdk/Adobe-Runtime-Support/discussions/1170#discussioncomment-7656524

**Reply by Jason0247** [reply]
Adding a transparent form to webView is a solution, but if there are too many animations or the size of the displayed objects is too large, efficiency can be greatly affected.
The grid language you mentioned is a good solution. Even without interaction, it can be used as a large-sized dynamic background. What properties are used to set this? Or do you directly use Bitmapdata for real-time rendering?

**Reply by itlancer** [reply]
Unfortunately I didn't have experience with such real-time rendering using `StageWebView`->`BitmapData`. But I suppose it could cause performance overhead for large areas.

**Comment by Jason0247** [comment]
Is there a way to draw webView as Bitmap?

**Reply by itlancer** [reply]
@Jason0247 
Take a look at `StageWebView::drawViewPortAsync()` https://airsdk.dev/reference/actionscript/3.0/flash/media/StageWebView.html#drawViewPortAsync()

**Reply by Jason0247** [reply]
Thank you.
